mesonbuild/dependencies/pkgconfig.py

```diff
@@ -78,38 +119,33 @@ def list_all(self) -> T.List[str]:
 class PkgConfigCLI(PkgConfigInterface):
     '''pkg-config CLI implementation'''
 
-    # The class's copy of the pkg-config path. Avoids having to search for it
-    # multiple times in the same Meson invocation.
-    class_pkgbin: PerMachine[T.Union[None, Literal[False], ExternalProgram]] = PerMachine(None, None)
-    # We cache all pkg-config subprocess invocations to avoid redundant calls
-    pkgbin_cache: T.Dict[
-        T.Tuple[ExternalProgram, T.Tuple[str, ...], T.FrozenSet[T.Tuple[str, str]]],
-        T.Tuple[int, str, str]
-    ] = {}
-
     def __init__(self, env: Environment, for_machine: MachineChoice, silent: bool) -> None:
-        self.env = env
-        self.for_machine = for_machine
-        # Store a copy of the pkg-config path on the object itself so it is
-        # stored in the pickled coredata and recovered.
-        self.pkgbin = self._detect_pkgbin(env, for_machine, silent)
+        super().__init__(env, for_machine)
+        self._detect_pkgbin()
+        if not silent:
+            if self.pkgbin:
+                mlog.log('Found pkg-config:', mlog.green('YES'), mlog.bold(f'({self.pkgbin.get_path()})'), mlog.blue(self.pkgbin_version))
+            else:
+                mlog.log('Found pkg-config:', mlog.red('NO'))
```

Comment:
I think this log should rather be in the `instance` method, since you may try different implementations before stating that `pkg-config` is not found.

Reply:
Good idea, fixed.